src/vtp/web/api/main.py

```diff
@@ -22,19 +41,63 @@ async def root() -> dict:
     return {"version": "0.1.0"}
 
 
+# Endpoint #1
 @app.get("/vote/")
 async def get_vote_store_id() -> dict:
     """Create and store a unique Vote Store ID for each client"""
-    vote_store_id = str(random.randrange(100000, 999999))
+
+    svdo = SetupVtpDemoOperation(
+        election_data_dir=DEFAULT_ELECTION_DATA_DIR,
+        verbosity=VERBOSITY,
+    )
+    guid = svdo.run(
+        guid_client_store=True,
+    )
+
+    vote_store_id = guid
     # add VoteStoreID to list
     vote_store_ids.append(vote_store_id)
     return {"VoteStoreID": vote_store_id}
 
 
+# Endpoint #2
 @app.get("/vote/{vote_store_id}")
 async def get_empty_ballot(vote_store_id: str) -> dict:
     """Return an empty ballot for a given Vote Store ID"""
+
+    if vote_store_id in vote_store_ids:
+        # Cet a (the) blank ballot from the backend
+        cbo = CastBallotOperation(
+            guid=vote_store_id,
+            verbosity=VERBOSITY,
+        )
+        empty_ballot = cbo.run(
+            blank_ballot=DEFAULT_BLANK_BALLOT,
+            return_bb=True,
+        )
+        return {"blank-ballot": empty_ballot}
+    return {"error": "VoteStoreID not found"}
+
+
+# Endpoint #3
+@app.get("/vote/cast-ballot/{vote_store_id}")
```

Comment:
Yes.  I figured that out but did not push that change because I got stuck with pydandic and typing trying to get the POST to work prior to commit'ing and pushing.

Reply:
Further reference: [RFC 9110 - GET](https://www.rfc-editor.org/rfc/rfc9110.html#name-get)

> Although request message framing is independent of the method used, content received in a GET request has no generally defined semantics, cannot alter the meaning or target of the request, and might lead some implementations to reject the request and close the connection because of its potential as a request smuggling attack ([Section 11.2](https://www.rfc-editor.org/rfc/rfc9112#section-11.2) of [[HTTP/1.1](https://www.rfc-editor.org/rfc/rfc9110.html#HTTP11)]). A client SHOULD NOT generate content in a GET request unless it is made directly to an origin server that has previously indicated, in or out of band, that such a request has a purpose and will be adequately supported. An origin server SHOULD NOT rely on private agreements to receive content, since participants in HTTP communication are often unaware of intermediaries along the request chain.